collector/src/main/java/io/prometheus/jmx/JmxScraper.java

```diff
@@ -160,9 +160,15 @@ private void scrapeBean(MBeanServerConnection beanConn, ObjectName mbeanName) {
             processAttributesOneByOne(beanConn, mbeanName, name2AttrInfo);
             return;
         }
-        for (Object attributeObj : attributes.asList()) {
-            if (Attribute.class.isInstance(attributeObj)) {
-                Attribute attribute = (Attribute)(attributeObj);
+        final List<Attribute> attributeList;
+        try {
+            attributeList = attributes.asList();
```

Comment:
The approach proposed is better than the current code, but still uses an "all-or-nothing" paradigm for a given MBean.

I feel a better approach would be to loop through the values in the `AttributeList`. If the Object is an instance of `Attribute`, process it... if it's not an `Attribute` log a message via a `logScrape` and ignore the object.